src/utils/useHotspot.tsx

```diff
@@ -0,0 +1,80 @@
+import { PublicKey } from '@solana/web3.js'
```

Comment:
I wonder if there would be any benefit in using redux here. Is there any benefit to caching this data? Seems it gets loaded in at least a couple places.